diff -u /include/jxl/jxl_export.h
--- /include/jxl/jxl_export.h
+++ /include/jxl/jxl_export.h
@@ -9,8 +9,7 @@

 #define JXL_EXPORT

-// TODO: go back to [[deprecated]]
-// https://github.com/libjxl/libjxl/issues/1388
-#define JXL_DEPRECATED __attribute__((__deprecated__))
+// MSVC requires [[deprecated]]
+#define JXL_DEPRECATED [[deprecated]]

 #endif /* JXL_EXPORT_H */
diff -u /src/lib/include/jxl/types.h /src/lib/include/jxl/types.h
--- /src/lib/include/jxl/types.h
+++ /src/lib/include/jxl/types.h
@@ -57,11 +57,11 @@ typedef enum {

 /* DEPRECATED: bit-packed 1-bit data type. Use JXL_TYPE_UINT8 instead.
  */
-static const int JXL_DEPRECATED JXL_TYPE_BOOLEAN = 1;
+JXL_DEPRECATED static const int JXL_TYPE_BOOLEAN = 1;

 /* DEPRECATED: uint32_t data type. Use JXL_TYPE_FLOAT instead.
  */
-static const int JXL_DEPRECATED JXL_TYPE_UINT32 = 4;
+JXL_DEPRECATED static const int JXL_TYPE_UINT32 = 4;

 /** Ordering of multi-byte data.
  */

