diff --git a/lib/xmlparse.c b/lib/xmlparse.c
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -453,16 +453,16 @@
   unsigned long long activationThresholdBytes;
 } ACCOUNTING;
 
-typedef struct entity_stats {
-/* BEGIN MOZILLA CHANGE (don't report debug information) */
+/* BEGIN MCP CHANGE (don't report debug information) */
 #if 0
+typedef struct entity_stats {
   unsigned int countEverOpened;
   unsigned int currentDepth;
   unsigned int maximumDepthSeen;
   unsigned long debugLevel;
-#endif
-/* END MOZILLA CHANGE */
 } ENTITY_STATS;
+#endif
+/* END MCP CHANGE */
 #endif /* XML_GE == 1 */
 
 typedef enum XML_Error PTRCALL Processor(XML_Parser parser, const char *start,
@@ -797,8 +797,12 @@
   unsigned long m_hash_secret_salt;
 #if XML_GE == 1
   ACCOUNTING m_accounting;
+/* BEGIN MCP CHANGE (don't report debug information) */
+#if 0
   ENTITY_STATS m_entity_stats;
 #endif
+/* END MCP CHANGE */
+#endif /* XML_GE == 1 */
 /* BEGIN MOZILLA CHANGE (Report opening tag of mismatched closing tag) */
   const XML_Char* m_mismatch;
 /* END MOZILLA CHANGE */
@@ -990,10 +994,14 @@
 
 static unsigned long
 ENTROPY_DEBUG(const char *label, unsigned long entropy) {
+/* BEGIN MCP CHANGE (don't report debug information) */
+#if 0
   if (getDebugLevel("EXPAT_ENTROPY_DEBUG", 0) >= 1u) {
     fprintf(stderr, "expat: Entropy: %s --> 0x%0*lx (%lu bytes)\n", label,
             (int)sizeof(entropy) * 2, entropy, (unsigned long)sizeof(entropy));
   }
+#endif
+/* END MCP CHANGE */
   return entropy;
 }
 
@@ -1324,12 +1332,12 @@
   parser->m_accounting.activationThresholdBytes
       = EXPAT_BILLION_LAUGHS_ATTACK_PROTECTION_ACTIVATION_THRESHOLD_DEFAULT;
 
-  memset(&parser->m_entity_stats, 0, sizeof(ENTITY_STATS));
-/* BEGIN MOZILLA CHANGE (don't report debug information) */
+/* BEGIN MCP CHANGE (don't report debug information) */
 #if 0
+  memset(&parser->m_entity_stats, 0, sizeof(ENTITY_STATS));
   parser->m_entity_stats.debugLevel = getDebugLevel("EXPAT_ENTITY_DEBUG", 0u);
 #endif
-/* END MOZILLA CHANGE */
+/* END MCP CHANGE */
 #endif
 }
 
